lighthouse-viewer/app/src/psi-api.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/** @typedef {{lighthouseResult?: LH.Result, error?: {message: string}}} PSIResponse */
+
+const PSI_URL = 'https://www.googleapis.com/pagespeedonline/v5/runPagespeed';
+const PSI_KEY = 'AIzaSyAjcDRNN9CX9dCazhqI4lGR7yyQbkd_oYE';
+const PSI_DEFAULT_CATEGORIES = [
+  'performance',
+  'accessibility',
+  'seo',
+  'best-practices',
+  'pwa',
+];
+
+/**
+ * Wrapper around the PSI API for fetching LHR.
+ */
+class PSIApi {
+  /**
+   * @param {PSIParams} params
+   * @return {Promise<PSIResponse>}
+   */
+  fetchPSI(params) {
+    params = Object.assign({
+      key: PSI_KEY,
```

Comment:
this is a little confusing and throws tsc for a loop (`key` isn't supposed to be on `params`), so what about just specific lines below of
`apiUrl.searchParams.append('key', PSI_KEY);`

and in the  loop
`if (key === 'strategy') value = value || 'mobile';`

or whatever.

(Ideally, if we ever upgrade off of `uglifyEs`, we could just do `const {strategy, category, ...restParams} = params;` and set those separately (and get a sweet `restParams` joke))

Reply:
> (Ideally, if we ever upgrade off of uglifyEs, we could just do const {strategy, category, ...restParams} = params; and set those separately (and get a sweet restParams joke))

Yeah, I had that at first. This marks the third time lack of a modern build has caused some pain.